[Pearl Twe] iP

peaol opened the pull request:
### iP

> Can't finish your assignments on time? Worry not, here's iP!

iP is a task tracker bot that helps prevent you from missing deadlines, it is:

- easy to use
- fast
- text-based

All you need to do is:
1. download it from [here](https://github.com/peaol/ip)
2. add your tasks

Features:

- [ ] Managing your tasks for you

- [ ] Interesting UI (Coming soon)

Sample code:
`public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }`

Commit:
Add Gradle support

peaol committed:
Level 1 implementation

peaol committed:
Level 2 implementation

peaol committed:
Level-3 implementation

peaol committed:
Level-4 Implementation

peaol committed:
Level-6 Implementation

peaol committed:
Level-5 Implementation after Level-6

peaol committed:
Cleaning up code after implementing all 6 levels

peaol committed:
A-TextUiTesting

peaol committed:
Add Level-8

peaol committed:
Improve abstraction in Duke class

peaol committed:
Add all classes into duke package

peaol committed:
Add more OOP

peaol committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S2/ip

peaol committed:
Add JUnit test cases

peaol committed:
Reconfigure test files

peaol committed:
Add JavaDoc comments to the code

peaol committed:
Adhere to coding standard

peaol committed:
Merge branch 'branch-A-CodingStandard'

bojie3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+}
+
+
+//        String logo = " ____        _        \n"
```
Possible to remove this logo since it would not be used and makes the code cleaner?

bojie3 commented on data/duke.txt:
```diff
@@ -0,0 +1,2 @@
+todo play true
```
data.txt should be empty since user would not have added any task initially when they start using Duke

bojie3 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.*;
```
Should refrain from the use of * in import statement.

bojie3 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+                String taskDescription = encoded[1];
+                String isDone = encoded[2];
+
+                switch (taskType) {
```
Nice use of switch statement

bojie3 reviewed (commented):
Very well done, great use oop principles and cod is easy to understand.

peaol committed:
Add find() method

peaol committed:
Merge branch 'branch-Level-9'

peaol committed:
Debugging an error found in Storage

peaol committed:
Add changes from tutor feedback

Zhongli5712 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,67 @@
+package duke.task;
+import java.util.List;
+import java.util.ArrayList;
+/**
+ * Abstract class Task for Todo, Deadline, and Event to inherit from
+ *
+ * @author Pearl Twe
+ * @version CS2103T AY22/23 Semester 2
+ */
```
I like that you have the the class description, however I think the description should follow the structure that start with a verb

Zhongli5712 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,67 @@
+        this.isDone = false;
+    }
+
+    /**
+     * Return description of task
+     * @return description
+     */
```
The verb should be in plural form, and there should be space between the description and @return line

Zhongli5712 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+import java.util.List;
+
+public class TaskList {
+    public List<Task> list;
```
should this be named "tasks" instead for further clarity?

Zhongli5712 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+        }
+    }
+
+    public void markTask(String[] parts) throws TaskDoesNotExistException {
+        //converting the index from String to Int
+        int indexInt = Integer.parseInt(parts[1]) - 1;
+
+        //in case the duke.task.Task index to mark exceeds current number of Tasks or neg number
+        if (indexInt + 1 > list.size() || indexInt < 0) {
+            throw new TaskDoesNotExistException();
+        }
+
+        Task t = list.get(indexInt);
+        t.markDone();
+        System.out.println("Nice! I've marked this duke.task as done:\n" + t.toString());
+    }
+
```
I like that you followed method naming convention and adding comment to explain the code

Zhongli5712 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,59 @@
+                    String[] parts = input.split(" ", 2);
+                    String taskType = parts[0];
+
+                    switch (taskType) {
+                        case "mark":
+                            taskList.markTask(parts);
+                            break;
+                        case "unmark":
+                            taskList.unmarkTask(parts);
+                            break;
+                        case "delete":
+                            taskList.deleteTask(parts);
+                            break;
+                        case "find":
+                            taskList.find(parts);
+                            break;
+                        case "todo":
+                        case "deadline":
+                        case "event":
+                            taskList.createTask(parts);
+                            break;
+                    }
```
there should not be indentation in switch case

Zhongli5712 reviewed (commented):
Overall, I found your code easy to follow, however there are still some minors details that you should notice. Well done!

peaol committed:
Add GUI to duke

peaol committed:
Add exit gui case

cmang12 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+     * Runs the user interface for abstraction
+     */
+    public void run() {
+        ui.greet();
```
I like that your code is short and follows SLAP.

cmang12 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,63 @@
+                if (input.equals("list")) {
+                    taskList.getList();
+                } else {
+                    //split into 2, first part is duke.task type, second part is instruction
```
I think the code format there is a little bit of arrowhead.

cmang12 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+                        if (isDone.equals("true")) {
+                            decodedEvent.markDone();
+                        }
+                        break;
```
Maybe the code can be abstracted even more to make it shorter?

cmang12 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+        list.remove(indexInt);
+        System.out.println("Noted. I've removed this duke.task: \n"
+                + task
+                + "\nNow you have " + list.size() + " tasks in the list.");
```
Perhaps you should let the print statement be handled by a different class?

cmang12 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,152 @@
+    public void deleteTask(String[] parts) throws TaskDoesNotExistException {
+        int indexInt = Integer.parseInt(parts[1]) - 1;
+        //in case the duke.task.Task index to mark exceeds current number of Tasks or neg number
+        if (indexInt + 1 > list.size() || indexInt < 0) {
```
Maybe you could also let the check index be handled by a different class like the parser

peaol committed:
Complete GUI

peaol committed:
Create Jar file

A jar zip file exists in out/artifacts/repos_jar

Anyone can download the jar file and run it in their cmd to access Duke

Refer to this tutorial on dealing with JAR files
https://se-education.org/guides/tutorials/gradle.html#creating-jar-files

peaol committed:
Improve GUI

peaol committed:
Add Assert statement

Assert statement added to Parser for better testing

So that it will notify the tester

peaol committed:
Merge pull request #2 from peaol/branch-A-Assertions

Add Assert statement to Parser class

peaol committed:
Improve code quality

Added changes to the variable names in TaskList class

To follow naming conventions and improve readability

peaol committed:
Merge pull request #3 from peaol/branch-A-CodeQuality

Improve code quality

peaol committed:
Add BCD extension

Added priority levels to task

For improved user experience

peaol committed:
Add User Guide on Github

peaol committed:
Release jar file

peaol committed:
Add better Gui

peaol committed:
Add Ui.png and user guide to docs
